File: 158.py
class Solution:
    """
    @param s: The first string
    @param b: The second string
    @return true or false
    """
    def anagram(self, s, t):

        def hashLize(s):
            dicts1= dict()
            for j in s:
                if j in dicts1.keys():
                    dicts1[j]+=1
                else:
                    dicts1[j]=0

            return dicts1

        # write your code here
        s1=hashLize(s)
        t1=hashLize(t)
        if s1 == t1:
            return True
        return False

File: 158.cpp
class Solution {
public:
    /**
     * @param s: The first string
     * @param b: The second string
     * @return true or false
     */

    typedef map<char,int> charHash;

    charHash keyHashLize(string s){
        	charHash s1;
        	for (int i=0;i<s.length();i++){
        		if (s1[s[i]]) s1[s[i]]++;
        		else s1[s[i]]=1;
        	}

        	return s1;
        }

    bool anagram(string s, string t) {
        // write your code here

    	

    	charHash s1=keyHashLize(s),t1=keyHashLize(t);
        if (s1==t1) return true;
        return false;
        
        }

        


    
};